src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1855,17 +2033,89 @@ bool isRefEscape
             ImmutableArray<RefKind> argRefKindsOpt,
             ImmutableArray<int> argsToParamsOpt,
             bool isRefEscape,
-            bool checkingMethodArgumentsMustMatch,
             bool ignoreArglistRefKinds,
-            ArrayBuilder<(ParameterSymbol? Parameter, BoundExpression Argument, bool IsRefEscape)> result)
+            ArrayBuilder<EscapeValue> escapeValues)
+        {
+            GetFilteredArguments(
+                symbol,
+                receiver,
+                parameters,
+                argsOpt,
+                argRefKindsOpt,
+                argsToParamsOpt,
+                ignoreArglistRefKinds,
+                mixableArguments: null,
+                escapeValues);
+
+            bool includeRefScopes = shouldIncludeRefScopes();
+            int index = 0;
+            while (index < escapeValues.Count)
+            {
+                var tuple = escapeValues[index];
+                if (!includeRefScopes && tuple.IsRefEscape)
+                {
+                    escapeValues.RemoveAt(index);
+                }
```

Comment:
Consider moving `includeRefScopes` and this check into `GetFilteredArguments()` to avoid adding to the list, then removing after.